File: procon20180520/backfront.py
n = int(input())
number_list = [[int(input())]]
max_length = 1
for i in range(n-1):
  temp = int(input())
  for nl in number_list:
    if nl[-1] == temp-1:
      nl.append(temp)
      length_size = len(nl)
      if length_size > max_length:
        max_length = length_size
      temp = 0
      break
  if temp != 0:
    number_list.append([temp])

print(n-max_length)

File: procon20180520/backfront.cpp
#include <iostream>

using namespace std;

int main() {
  int n;
  int arr[400000][400000] = {};
  int temp;
  int max_length = 1;
  int current_position = 0;
  cin >> n;
  cin >> temp;
  arr[0][0] = temp;
  for(int i = 0; i< n-1; i++){
    cin >> temp;
    for(int j = 0; j<n; j++){
      for(int k = 0; k<n; k++){
        if(arr[j][k]==0){
          if(arr[j][k-1] == temp-1){
            arr[j][k] = temp;
          }
          if(k+1>max_length){
            max_length = k+1;
          }
          temp = 0;
          break;
        }
      }
      if(temp == 0){
        break;
      }
    }
    if(temp != 0){
      arr[current_position+1][0] = temp;
      current_position++;
    }
  }
  cout << arr << endl;
}
